src/main/java/org/dmfs/jems/single/elementary/Reduced.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.jems.single.elementary;
+
+import org.dmfs.jems.function.BiFunction;
+import org.dmfs.jems.single.Single;
+
+
+/**
+ * The {@link Single} value of a reduced {@link Iterable}.
+ * <p>
+ * Note, this is the single threaded version of a "reduce" operation.
+ *
+ * @author Marten Gajda
+ */
+public final class Reduced<V, T> implements Single<T>
+{
+    private final T mInitialValue;
+    private final Iterable<V> mIterable;
+    private final BiFunction<V, T, T> mFunction;
+
+
+    public Reduced(T initialValue, BiFunction<V, T, T> function, Iterable<V> iterable)
```

Comment:
Just a questions, wouldn't it look better if the `BiFunction` has type parameters `T, V, T`, so the the aggregate type comes first in the method? And maybe give some name to the type parameters? As I see the function is called 'accumulator' elsewhere.

Reply:
Right. I updated the code.